lib/internal/test_runner/test.js

```diff
@@ -548,7 +539,10 @@ class Test extends AsyncResource {
       const runArgs = ArrayPrototypeSlice(args);
       ArrayPrototypeUnshift(runArgs, this.fn, ctx);
 
-      if (this.fn.length === runArgs.length - 1) {
+      if (this.fn === noop && this.parent === null) {
+        // Avoid dealy of microtasks for root test
+        // to allow calling root.run() after event loop has ended
```

Comment:
This feels kind of strange. Do we need to do this if we await the exit handler in harness.js?

Reply:
thanks, it is not needed now